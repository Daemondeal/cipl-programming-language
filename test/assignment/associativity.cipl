let a = "a"
let b = "b"
let c = "c"

a = b = c

print(a)
print(b)
print(c)