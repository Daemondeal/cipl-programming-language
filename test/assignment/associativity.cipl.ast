(let a a)
(let b b)
(let c c)
(expression (assign a (assign b c)))
(expression (call print a))
(expression (call print b))
(expression (call print c))

