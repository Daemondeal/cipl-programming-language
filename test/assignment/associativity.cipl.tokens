Token          Lexeme         Line 
LET            let            1    
IDENTIFIER     a              1    
EQUAL          =              1    
STRING         "a"            1     a
LINE_END       \n             2    
LET            let            2    
IDENTIFIER     b              2    
EQUAL          =              2    
STRING         "b"            2     b
LINE_END       \n             3    
LET            let            3    
IDENTIFIER     c              3    
EQUAL          =              3    
STRING         "c"            3     c
LINE_END       \n             4    
IDENTIFIER     a              5    
EQUAL          =              5    
IDENTIFIER     b              5    
EQUAL          =              5    
IDENTIFIER     c              5    
LINE_END       \n             6    
IDENTIFIER     println        7    
LEFT_PAREN     (              7    
IDENTIFIER     a              7    
RIGHT_PAREN    )              7    
LINE_END       \n             8    
IDENTIFIER     println        8    
LEFT_PAREN     (              8    
IDENTIFIER     b              8    
RIGHT_PAREN    )              8    
LINE_END       \n             9    
IDENTIFIER     println        9    
LEFT_PAREN     (              9    
IDENTIFIER     c              9    
RIGHT_PAREN    )              9    
LINE_END       \n             9    
EOF                           9    
