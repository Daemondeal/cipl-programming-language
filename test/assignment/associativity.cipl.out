::stdout::
c
c
c

::stderr::
